### PR TITLE
fix: provider stylesheet inheritance and display format

### PR DESCRIPTION
**Pull request by InteractiveTimmy**
# Description

The scope of the changes in this pull request are to update the `display` of the `<mdc-themeprovider />` component to be `contents`, [which should be used for non-block-impacting components](https://developer.mozilla.org/en-US/docs/Web/CSS/display#contents).

Additionally, Stylesheet inheritance was being misrepresented from the `Provider` model into the `ThemeProvider` class definition.

Storybooks updated accordingly

# Links

* [Developer documentation on display contents]((https://developer.mozilla.org/en-US/docs/Web/CSS/display#contents).)

### CODE REVIEW

File: packages/components/src/components/themeprovider/themeprovider.component.ts [hunk 1]
```diff
@@ -95,7 +96,7 @@ class ThemeProvider extends Provider<ThemeProviderContext> {
     }
   }
 
-  public static override styles = styles;
+  public static override styles: Array<CSSResult> = [...Provider.styles, styles];
```
Comment (chrisadubois) [inline packages/components/src/components/themeprovider/themeprovider.component.ts:99]: good catch

File: packages/components/src/models/provider/provider.component.ts [hunk 2]
```diff
@@ -53,7 +53,7 @@ abstract class Provider<C> extends Component {
   /**
    * Styles associated with this Provider Component.
    */
-  public static override styles: CSSResult | Array<CSSResult> = styles;
+  public static override styles: Array<CSSResult> = [styles];
```
Comment (chrisadubois) [inline packages/components/src/models/provider/provider.component.ts:56]: noice

File: packages/components/src/models/provider/provider.styles.ts [hunk 3]
```diff
@@ -2,12 +2,7 @@ import { css } from 'lit';
 
 const styles = css`
   :host {
-    border: 0;
-    display: block;
-    height: 100%;
-    margin: 0;
-    padding: 0;
-    width: 100%;
+    display: contents;
```
Comment (chrisadubois) [inline packages/components/src/models/provider/provider.styles.ts:5]: 👍

**Review by chrisadubois — APPROVED** [review]
lgtm, but would want @cscheffauer approval.

tests are also failing.

File: packages/components/src/components/iconprovider/iconprovider.styles.ts [hunk 4]
```diff
@@ -0,0 +1,11 @@
+import { css } from 'lit';
+
+const styles = css`
```
Comment (cscheffauer) [inline packages/components/src/components/iconprovider/iconprovider.styles.ts:3]: @Jason20020 this is against what Tim did with his change - why exactly did you need to set this? Where the screenshots off? or what is the impact if we don't set this style here?

**Review by cscheffauer — CHANGES_REQUESTED** [review]

**Review by cscheffauer — APPROVED** [review]

**Pull request merged**